src/Discord/Parts/User/Member.php

```diff
@@ -305,8 +306,13 @@ public function sendMessage($message, bool $tts = false, $embed = null, $allowed
      *
      * @return RolePermission
      */
-    public function getPermissions(?Channel $channel = null): RolePermission
+    public function getPermissions($channel = null): RolePermission
     {
+        if ($channel instanceof thread) {
+            $channel = $this->guild->channels->get('id', $channel->parent_id);
+        } elseif (!($channel instanceof channel)) {
+           return $this->factory->part(RolePermission::class, ['bitwise' => 0]);
+        }
```

Comment:
This will prevent `$channel` being passed as `null`. Regardless, I think it should be changed to throw an exception in the case `$channel` is not `Channel`, `Thread` or `null`.